### PR TITLE
Restructuring, Adding Import and Export

### PR DESCRIPTION
**Pull request by mrwiora**
Hi Malte! :)

I've restructured your code and added an import and export functionality.
It would be great if you could review my code changes.

Cheers,
Matthias

### CODE REVIEW

**Comment by derBroBro** [comment]
Hello Matthias,

thank you for the large update and the new import / export function. I try to check the code this week. :-)

Best Malte

**Comment by mrwiora** [comment]
Hi Malte,

thanks! :)
I just added some s3-capabilities (storing the user-role-cookie-content).

Cheers,
Matthias

File: LICENSE.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 The MIT License (MIT)
 
 Copyright (c) 2016 Malte Brodersen
+Copyright (c) 2017 Matthias R. Wiora
```
Comment (derBroBro) [inline LICENSE.txt:4]: Unusual, we can add a part to the readme.md who contributed which feature.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+[![Code Climate](https://codeclimate.com/github/mwiora/aws-role-editor/badges/gpa.svg)](https://codeclimate.com/github/mwiora/aws-role-editor)
```
Comment (derBroBro) [inline README.md:1]: Needs to be set to this target project.

File: README.md [hunk 3]
```diff
@@ -18,17 +20,33 @@ Because many people, including myself, are very sensitive using third party tool
...
+    "accessKeyId":"ACCESSKEYID",
+    "secretAccessKey":"SECRET/ACCESS/KEY",
+    "region":""
+}
+```
+
 ### Screenshots
 #### Editor
-![example image](https://lh3.googleusercontent.com/_uWwbvqql457gLc2oDq5-vJ6b8pLRTkfdXDC8RkUlRUCDv4WdyhNjVMZ-OGUfyaCBPLszAwQ=s640-h400-e365-rw "AWS Role Editor")
+![Editor](https://raw.githubusercontent.com/mwiora/aws-role-editor/master/res/screen02.png "AWS Role Editor")
```
Comment (derBroBro) [inline README.md:50]: Needs to be set to this target project.

File: README.md [hunk 4]
```diff
@@ -18,17 +20,33 @@ Because many people, including myself, are very sensitive using third party tool
...
 #### As displayed in the AWS Console
-![example image](https://lh3.googleusercontent.com/dhTb_FYvg7A07BR4SOR1Oopi6quBxDe-XY0iaje9aptNei-ZDhE83AGp1hqzN7juEA0JjcDV=s640-h400-e365-rw "AWS Role Editor")
+![AWS Console](https://raw.githubusercontent.com/mwiora/aws-role-editor/master/res/screen01.png "AWS Role Editor")
```
Comment (derBroBro) [inline README.md:52]: Needs to be set to this target project.

File: js/popup.js [hunk 5]
```diff
@@ -1,116 +1,258 @@
...
+    try {
+        JSON.parse(str);
+    } catch (e) {
+        return false;
+    }
+        return true;
+    }
+
+    // function to access s3 based on the informations provided in the local json file - TODO: calls drawform, but should only return the cookie...
+    function getConfigFromS3(accessKeyId, secretAccessKey, region, s3bucket, s3key) {
```
Comment (derBroBro) [inline js/popup.js:239]: I tried to keep such information out of the extension to keep it lean for security reasons.
Mostly the enterprises I know, have many more then five accounts.
An option would to provide a "import from URL" function. Here a presigned S3-URL can be provided which is much more secure then create a whole IAM user.
Also a GIST or other site can be used.

**Comment by derBroBro** [comment]
Hi Matthias,

I added some comments.
It would be greate if we can try to keep the PR as small as possible and create new ones if needed. Each should only contain one feature. Specialy the S3 option is such a new feature which would great to discuss in an extra scope.

Best Malte

**Review by derBroBro — CHANGES_REQUESTED** [review]
Hi Matthias,

I added some comments.
It would be greate if we can try to keep the PR as small as possible and create new ones if needed. Each should only contain one feature. Specialy the S3 option is such a new feature which would great to discuss in an extra scope.

Best Malte

**Pull request closed without merging**